### ISSUE TITLE
How to connect to a secure flamserver?

### ISSUE THREAD

**Issue by krey** [irmen/Pyro4#176]
Hi,

How to connect to a secure flameserver?

Server side: python -m Pyro4.utils.flameserver -H 192.168.1.102 -p 64444 -k 123456

Client side: flame=Pyro4.utils.flame.connect("192.168.1.102:64444, key=123456") ?

I want explanation for connecting on a server with HMAC_Key

I found this reference, but it no longer works: http://pyro-core.narkive.com/GxndCV2C/hmac-key

Thanks,

krey

**Comment by irmen** [comment]
Since Pyro 4.30 you have to set the proxy._pyroHmacKey on a per-proxy basis.  This is not exposed through the flame.connect utility function, unfortunately. I think this is an easy fix for the next release.

For the time being, you can however do it yourself easily by creating a flame proxy directly instead of using the connect utility function:

```
flame = Pyro4.Proxy("PYRO:Pyro.Flame@localhost:49812")   # use whatever URI the server prints when started
flame._pyroHmacKey = b"123456"
flame.evaluate("3+4")
```

**Comment by irmen** [comment]
fixed for 4.61 by 09b846c3b588a000947abeda6fa8981728fde5e3
